Source/WebInspectorUI/UserInterface/Controllers/JavaScriptLogViewController.js

```diff
@@ -369,13 +369,33 @@ WI.JavaScriptLogViewController = class JavaScriptLogViewController extends WI.Ob
                 var groupElement = group.render(messageView);
                 this._currentSessionOrGroup.append(groupElement);
                 this._currentSessionOrGroup = group;
-            } else
-                this._currentSessionOrGroup.addMessageView(messageView);
+            } else this._currentSessionOrGroup.addMessageView(messageView);
+            
+            if (WI.settings.experimentalGroupSourceMapErrors.value && WI.consoleManager.failedSourceMapConsoleMessages.has(messageView.message))
+                this._addConsoleMessageToConsoleSourceMapErrorGroup(messageView);
         }
 
         if (this.delegate && typeof this.delegate.didAppendConsoleMessageView === "function")
             this.delegate.didAppendConsoleMessageView(messageView);
     }
+    
+    _addConsoleMessageToConsoleSourceMapErrorGroup(messageView)
+    {
+            
+        if (this._firstSourceMapErrorMessageView === undefined) {
+            this._firstSourceMapErrorMessageView = messageView;
+            return;
+        }
+        
+        if (this._failedSourceMapsGroup === undefined) {
+            let currentSession = this._sessions.lastValue;
```

Comment:
why not `this._currentSessionOrGroup`?

Reply:
This is an intentional decision to avoid the `ConsoleSourceMapMessageGroup` from being added to an opened ConsoleGroup. As a special case of grouped errors, we want to ensure it is always top-level and not unintentionally hidden by being nested in a collapsed group whether created by us or by the user. cc @rcaliman-apple

Reply:
> As a special case of grouped errors

These are resource loading errors that pertain to the main frame (and subframes) so we don't want to have them accidentally hidden inside an unrelated `ConsoleGroup`.

Reply:
i see.  so you're specifically trying to get the current top-level group.  in that case i'd rename this to `currentTopLevelSession` or something like that to better indicate the distinction

Reply:
FYI im still seeing the name as `currentSession`